Source/WebCore/animation/KeyframeEffect.cpp

```diff
@@ -1223,23 +1222,28 @@ const String KeyframeEffect::pseudoElement() const
 
     // The target pseudo-selector. null if this effect has no effect target or if the effect target is an element (i.e. not a pseudo-element).
     // When the effect target is a pseudo-element, this specifies the pseudo-element selector (e.g. ::before).
+    // FIXME: This needs proper serialization for name arguments.
     if (targetsPseudoElement())
-        return pseudoIdAsString(m_pseudoId);
+        return pseudoIdAsString(m_pseudoElementIdentifier->pseudoId);
     return { };
 }
 
 ExceptionOr<void> KeyframeEffect::setPseudoElement(const String& pseudoElement)
 {
     // https://drafts.csswg.org/web-animations-1/#dom-keyframeeffect-pseudoelement
+    // FIXME: This needs proper conversion for name arguments.
     auto pseudoId = pseudoIdFromString(pseudoElement);
     if (!pseudoId)
         return Exception { ExceptionCode::SyntaxError, "Parsing pseudo-element selector failed"_s };
 
-    if (*pseudoId == m_pseudoId)
+    if (!m_pseudoElementIdentifier && pseudoId == PseudoId::None)
+        return { };
+
+    if (m_pseudoElementIdentifier && *pseudoId == m_pseudoElementIdentifier->pseudoId)
         return { };
 
     auto& previousTargetStyleable = targetStyleable();
-    m_pseudoId = *pseudoId;
+    m_pseudoElementIdentifier = pseudoId == PseudoId::None ? std::nullopt : std::optional(Style::PseudoElementIdentifier { *pseudoId });
```

Comment:
Wish we had a helper function for this. It’s a verbose pattern and occurs at least two times, maybe more.

Reply:
This will be gone once we fix more code to use `std::optional<PseudoElementIdentifier>`